### ISSUE TITLE
Usage of emission model

### ISSUE THREAD

**Issue by bargisou** [ManiAm/VENTOS_Public#33]
Is it possible to collect vehicle's emissions like collecting vehicle's data for "platoon_cacc" project?
I created a text file named "000_vehicleEmissions.txt" in the results folder.But emission file dose'nt seem to update like the vehicledata.txt file.

**Comment by ManiAm** [comment]
collecting emission is per vehicle. Have you checked section '29.4 Collect Vehicle’s Emission' ?

You need to use something like this in the ini configuration for every vehicle that you need to collect emission from:

Network.V[0].record_emission = true
Network.V[0].emission_list = "vehId | emissionClass | CO2 | CO | HC | NOx"

**Comment by bargisou** [comment]
Yes.I am using it the same way for per vehicle but the text file seems to be empty even after simulation.
Are there any other steps than the above mentioned to create the record list?
Is it possible to measure fuel inside ventos?

**Comment by bargisou** [comment]
Is it necessary to create an empty file named 000_vehicleEmission.txt  in the results folder along with 000_vehicledata.txt file prior to simulation, for values to update in it .Or after adding above mentioned two lines in the omnetpp.ini file the file automatically generates?
Is there any way to record fuel consumption also along with other data?

**Comment by zeroenergyhere** [comment]
Same error, emission data are not logged after adding Network.V[0].record_list=" ..... " (as per documentation)". 
After re-installation Simulation terminated with exit code: 1, environment and path error, even though i have added the path of omnetpp.

**Comment by ManiAm** [comment]
>> Is it necessary to create an empty file named 000_vehicleEmission.txt in the results folder 

No. That file will be created automatically
No need to create it manually.
Are you sure that the simulation completes without any errors?
If the simulation terminates with error no output txt file will be written.

I have tested a scenario myself and it is working.

Network.V[*].record_emission = true
Network.V[*].emission_list = "vehId | emissionClass | CO2 | CO | HC | NOx"

![2018-05-30_8-39-43](https://user-images.githubusercontent.com/5473998/40731110-37bd9f88-63e5-11e8-89e8-e203c09c6a2c.png)

**Comment by ManiAm** [comment]
>> After re-installation Simulation terminated with exit code: 1, environment and path error, even though i have added the path of omnetpp.

This has nothing to do with emission recording. If the simulation cannot find the SUMO path then you have not followed the installation manual correctly. Check Step 4 in section '2 Installing VENTOS' of the manual.

**Comment by bargisou** [comment]
Thank you for your reply....The emission recording works properly....Is there any possibility to record fuel consumption also in the same way as that of emissions?

**Comment by ManiAm** [comment]
>> Is there any possibility to record fuel consumption also in the same way as that of emissions?

As shown in section '29.4 Collect Vehicle’s Emission' of the manual, you can add fuel into the string.

```
Network.V[*].record_emission = true
Network.V[*].emission_list = "vehId | emissionClass | CO2 | CO | HC | NOx | fuel"
```
basically adding words to the emission_list causes the simulation to collect that for you (you can remove an emission if you do not need it or you can change the order of them to change the order of column in the txt file)

**Comment by bargisou** [comment]
Thank you once again.With the above mentioned changes I am able to collect the required data.
